### ISSUE TITLE
Issue registering projectors - Cannot declare class

### ISSUE THREAD

**Issue by anabeto93** [spatie/laravel-event-sourcing#301]
In my `composer.json` I have created a custom namespace which may be necessary for this bug.

```bash
{
        "psr-4": {
            "App\\": "app/",
            "Domain\\": "app/Domain/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        },
        "files": [
            "app/helpers.php"
        ]
    }
```

I also have a Projector called `MerchantActivityProjector` but when I try registering it in `event-sourcing.php`, my entire application fails throwing an error ` Cannot declare class Domain\Merchant\Projectors\MerchantActivityProjector, because the name is already in use`

I first noticed this when I was trying to use a ServiceProvider, `EventSourcingServiceProvider` with a Projectionist to register the projectors and it gave similar errors so I reverted to using `event-sourcing.php` which I've used before in v4 without problems.

**Comment by anabeto93** [comment]
I just noticed auto-discovery has been my problem all along. I still think this is a bug worth looking at.

Auto discovery seems to work? as in discover projectors but it doesn't really work. 
![manual_registration_works_not_auto_discovery](https://user-images.githubusercontent.com/7607657/146861169-0898c78f-3a66-4e9f-bb70-dbb674465a2a.png)

Projector doesn't work by default if it is auto_discovered and it throws the above error `Cannot declare class ${ProjectorClassNameHere}` if you try registrating a projector in `projectors` of `event-sourcing.php`. But if you turn off auto discovery and manually register the projectors, they work.

I didn't have to comment out `app()->path()` under `auto_discover_projectors_and_reactors` section in `v4` but I had to manually register projectors before it worked.

Sample test success response.
![event_sourcing_works](https://user-images.githubusercontent.com/7607657/146861803-760da812-a4e5-4ba7-8aa3-68c335d17c77.png)
